sklearn/model_selection/split.py

```diff
@@ -0,0 +1,1106 @@
+"""
+The :mod:`sklearn.model_selection.split` module includes classes and
+functions to split the data based on a preset strategy.
+"""
+
+# Author: Alexandre Gramfort <alexandre.gramfort@inria.fr>,
+#         Gael Varoquaux <gael.varoquaux@normalesup.org>,
+#         Olivier Girsel <olivier.grisel@ensta.org>
+# License: BSD 3 clause
+
+
+from __future__ import print_function
+from __future__ import division
+
+import warnings
+from itertools import chain, combinations
+from math import ceil, floor, factorial
+import numbers
+from abc import ABCMeta, abstractmethod
+
+import numpy as np
+import scipy.sparse as sp
+
+from ..utils import indexable, check_random_state, safe_indexing
+from ..utils.validation import _num_samples, check_array, column_or_1d
+from ..utils.multiclass import type_of_target
+from ..externals.six import with_metaclass
+from ..externals.six.moves import zip
+from ..utils.fixes import bincount
+
+
+__all__ = ['KFold',
+           'LeaveOneLabelOut',
+           'LeaveOneOut',
+           'LeavePLabelOut',
+           'LeavePOut',
+           'ShuffleSplit',
+           'StratifiedKFold',
+           'StratifiedShuffleSplit',
+           'PredefinedSplit',
+           'train_test_split',
+           'check_cv']
+
+
+class _PartitionIterator(with_metaclass(ABCMeta)):
```

Comment:
Why doesn't this have a generic ``__repr__`` similar to ``BaseEstimator``? Then we could save all the effort below, right?